repos/system_upgrade/el7toel8/actors/checkosrelease/libraries/library.py

```diff
@@ -0,0 +1,59 @@
+import os
+
+from leapp.exceptions import StopActorExecution, StopActorExecutionError
+from leapp.libraries.common import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import OSReleaseFacts
+
+
+def skip_check():
+    """ Check if has environment variable to skip this actor checks """
+    if os.getenv('LEAPP_SKIP_CHECK_OS_RELEASE'):
+        reporting.report_generic(
+            title='Skipped OS release check',
+            severity='low',
+            summary='OS release check skipped via LEAPP_SKIP_CHECK_OS_RELEASE env var.',
+        )
+        return True
+    return False
+
+
+def check_os_version(supported_version):
+    """ Check OS version and inhibit upgrade if not the same as supported ones """
+    if not isinstance(supported_version, dict):
+        api.current_logger().warning('The supported version value is invalid.')
+        raise StopActorExecution()
+
+    facts_messages = api.consume(OSReleaseFacts)
+    facts = next(facts_messages, None)
+    if list(facts_messages):
+        api.current_logger().warning('Unexpectedly received more than one OSReleaseFacts message.')
+    if not facts:
+        raise StopActorExecutionError(
+            'Could not check OS version', details={'details': 'No OSReleaseFacts facts found.'}
+        )
+
+    if facts.release_id not in supported_version:
+        reporting.report_generic(
+            title='Unsupported OS id',
+            summary='Supported OS ids for upgrade process: {}'.format(', '.join(supported_version.keys())),
+            flags=['inhibitor'],
+        )
+        return
+
+    if not isinstance(supported_version[facts.release_id], list):
+        reporting.report_generic(
```

Comment:
This is not a problem to be solved by the customer, it makes no sense to add it to report IMO. Customers have no way to provide a valid list. This is an internal problem, so I'd end the leapp execution using one of the exceptions (StopActorExecution).